clientconn.go

```diff
@@ -996,9 +996,7 @@ func (ac *addrConn) resetTransport() {
 			connectDeadline := time.Now().Add(dialDuration)
 
 			ac.mu.Lock()
-			ac.cc.mu.RLock()
```

Comment:
Actually, this read lock is correct. It protects the reading of `ac.cc.mkp`. We should keep it.

The problem is the writing of `ac.dopts.copts.KeepaliveParams`. It needs to hold `ac.mu`.
In the older version mentioned in the issue, the writing happens outside of `ac.mu` (`unlock` before this line, `lock` after this line).
https://github.com/grpc/grpc-go/blob/97da9e087c5cf7a11263637efb410e08c51a7a0f/clientconn.go#L904-L914

But this is already fixed in the new version
https://github.com/grpc/grpc-go/blob/c71aa62423b37215980f9c3141eef06f4c35e998/clientconn.go#L997-L1002

So I think the race only exists in the old version, and the changes here are all unnecessary.